### ISSUE TITLE
Can not detect object using inference.py

### ISSUE THREAD

**Issue by longann890621** [NVlabs/Deep_Object_Pose#411]
Hi, here is the problem I encountered, I would like to ask you for advice:
I used pose_generation.py and dope_config.yaml to generate 3000 synthetic data with isaacsim.

[dope_config.txt](https://github.com/user-attachments/files/21033407/dope_config.txt)

[pose_generation.txt](https://github.com/user-attachments/files/21033377/pose_generation.txt)

Then I divided the data into train, val, test (0.8, 0.1, 0.1), and trained it with train.py.
Training instructions:
python3 train.py \
  --data ~/AI_Nvidia/Isaac_Sim/output_split/train \
  --object _03_cracker_box _35_power_drill \
  --epochs 50 \
  --batchsize 8 \
  --outf ~/AI_Nvidia/Isaac_Sim/output_split/training_output  

After the training is completed, I got final_net_epoch_0050.pth and some training results graphs.

![Image](https://github.com/user-attachments/assets/e6c13e46-76f9-4681-b953-6258f9683b0f)

![Image](https://github.com/user-attachments/assets/aca9ee85-d6ce-4299-a63b-f2047e182494)

Then I used inference.py and config_pose.yaml and camera_info.yaml to make inferences on my test dataset.

[inference.txt](https://github.com/user-attachments/files/21033512/inference.txt)

[config_pose.txt](https://github.com/user-attachments/files/21033530/config_pose.txt)

[camera_info.txt](https://github.com/user-attachments/files/21033536/camera_info.txt)

Inference instructions:
python3 inference.py \
  --data /home/robertlo/AI_Nvidia/Isaac_Sim/output_split/test \
  --weights /home/robertlo/AI_Nvidia/Isaac_Sim/output_split/training_output/final_net_epoch_0050.pth \
  --config ../config/config_pose.yaml \
  --object _03_cracker_box \
  --camera ../config/camera_info.yaml \
  --outf /home/robertlo/AI_Nvidia/Isaac_Sim/output_split/test_predictions/cracker_box \
  --debug
  
After prediction, I get an empty json file and belief_maps.png. It seems that the inference failed!
I would like to ask if I did something wrong?

![Image](https://github.com/user-attachments/assets/8115531d-a389-4a77-a5b3-3a74a1c0b21a)

![Image](https://github.com/user-attachments/assets/be33497a-19dd-47a0-ac9d-863767321566)

**Comment by longann890621** [comment]
Hi, first of all, thank you for your reply!
I have changed my dope_config.yaml as you said：#- { part_name: 035_power_drill, num: 1, prim_type: _35_power_drill }

[dope_config.txt](https://github.com/user-attachments/files/21051448/dope_config.txt)

[pose_generation.txt](https://github.com/user-attachments/files/21051604/pose_generation.txt)

Then I generated 5,000 synthetic data and tried to run train.py for training with a small amount of data.

[train.txt](https://github.com/user-attachments/files/21051462/train.txt)

Training instructions:
python3 train.py \
  --data ~/AI_Nvidia/Isaac_Sim/output_split/train \
  --object _03_cracker_box \
  --epochs 300 \
  --batchsize 8 \
  --save_every 30 \
  --outf ~/AI_Nvidia/Isaac_Sim/output_split/training_output
  
Halfway through the training I discovered a problem:

[Terminal report.txt](https://github.com/user-attachments/files/21051528/Terminal.report.txt)

The following frequently appears in the reports printed from the terminal：NaN or Inf found in input tensor.
Is this normal? If not, how can I correct it?

**Comment by longann890621** [comment]
Hello, after referring to your suggestions, I did a long training to get the weight file, but some problems occurred during inference:

1.
I used your example program pose_generation.py and the example program dope_config.yaml with only the training object changed.

[pose_generation.txt](https://github.com/user-attachments/files/21208305/pose_generation.txt)

[dope_config.txt](https://github.com/user-attachments/files/21208307/dope_config.txt)

Use the following command to generate training data:
./python.sh standalone_examples/replicator/pose_generation/pose_generation.py --num_mesh 0 --num_dome 20000

I divide the generated data into train, test (0.9, 0.1)
Then I used --epochs 500 --batchsize 32 for training. The training results are shown in the figure.

<img width="1981" height="988" alt="Image" src="https://github.com/user-attachments/assets/ef827ba1-3008-45d6-ab47-ce045dcae0ec" />

<img width="1981" height="988" alt="Image" src="https://github.com/user-attachments/assets/8e368473-7c33-4c42-a614-1e48aecb4843" />

The loss has dropped to 0.0011, why is there only a 21% prediction success rate for the test dataset generated in the same way? (420 out of 2000 photos were inferred successfully)
I want to know if the way I prepared the dataset is correct? In addition, is it normal that the pose prediction cannot be made if the object position exceeds the image? I found that as long as the object exceeds the image, the inference will fail.

2.
The scenario I want to infer is as follows:

<img width="512" height="512" alt="Image" src="https://github.com/user-attachments/assets/4a90ac06-7924-4835-928b-e4bd22c37b9a" />

Can the weights I trained using the photos generated by pose_generation.py be adapted to this simple environment?
And, I got the following weight file:

<img width="1502" height="652" alt="Image" src="https://github.com/user-attachments/assets/223af890-882a-4ae3-b0a7-2db64ec3a697" />

net_epoch_0200.pth can infer some of these scenarios, but net_epoch_0500.pth cannot infer such scenarios, and final_net_epoch_0500.pth cannot infer such scenarios. Why can't we make effective inferences after training to later weights? Will final_net_epoch_XXXX.pth select the best result? Or is it just a temporary archive of the process?
